src/platform/unittest/ToeplitzTest.cpp

```diff
@@ -0,0 +1,194 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+--*/
+
+#define _CRT_SECURE_NO_WARNINGS 1
+#include "main.h"
+#include "msquic.h"
+#include "msquichelper.h"
+#include "quic_toeplitz.h"
+#include <stdio.h>
+
+#ifdef QUIC_CLOG
+#include "ToeplitzTest.cpp.clog.h"
+#endif
+
+struct ToeplitzTest : public ::testing::Test
+{
+    protected:
+    static const char* HashKey;
+
+    struct QuicBuffer
+    {
+        uint8_t* Data;
+        uint16_t Length;
+
+        QuicBuffer(const char* HexBytes)
+        {
+            Length = (uint16_t)(strlen(HexBytes) / 2);
+            Data = new uint8_t[Length];
+
+            for (uint16_t i = 0; i < Length; ++i) {
+                Data[i] =
+                    (DecodeHexChar(HexBytes[i * 2]) << 4) |
+                    DecodeHexChar(HexBytes[i * 2 + 1]);
+            }
+        }
+
+        QuicBuffer(const QuicBuffer&) = delete;
+        QuicBuffer(QuicBuffer&&) = delete;
+
+        ~QuicBuffer()
+        {
+            delete [] Data;
+        }
+    };
+
+    struct QuicTestAddress {
+        QUIC_ADDR Addr;
+        QuicTestAddress(const char* AddrStr, uint16_t Port) {
+            EXPECT_TRUE(QuicAddrFromString(AddrStr, Port, &Addr));
+        }
+    };
+
+    static
+    auto
+    ValidateRssToeplitzHash(
+        _In_ const char** ExpectedHashes,
+        _In_ const char** SourceAddresses,
+        _In_ const uint16_t* SourcePorts,
+        _In_ const char** DestinationAddresses,
+        _In_ const uint16_t* DestinationPorts,
+        _In_ uint32_t TestCaseCount,
+        _In_ QUIC_ADDRESS_FAMILY Family
+        )
+    {
+        const QuicBuffer KeyBuffer(HashKey);
+
+        CXPLAT_TOEPLITZ_HASH ToeplitzHash{};
+        CxPlatCopyMemory(ToeplitzHash.HashKey, KeyBuffer.Data, KeyBuffer.Length);
+        ToeplitzHash.InputSize = CXPLAT_TOEPLITZ_INPUT_SIZE_IP;
+        CxPlatToeplitzHashInitialize(&ToeplitzHash);
+
+        for (uint32_t i = 0; i < TestCaseCount; i++) {
```

Comment:
nit:
I think the loop should be kept in the caller (or no loop, and several calls with each test cases).
It adds complexity to understand the test, keeping track of 5 arrays, etc...

The test case could read:

```
ValidateRssToeplitzHash("foo", "bar", "baz", ...);
ValidateRssToeplitzHash("foo2", "bar2", "baz2, ...);
...
```

otherwise, which is both simpler, shorter, and removes room from mis-counting errors.
If the initialization can't be repeated, it could be split in a "prepare" function.